src/overlays/actors/ovl_Obj_Mure2/z_obj_mure2.c

```diff
@@ -30,43 +51,204 @@ ActorInit Obj_Mure2_InitVars = {
+    OBJ_MURE2_CHILD_COUNT_BUSH_RING,      // OBJMURE2_CHILDTYPE_BUSH_RING
+    OBJ_MURE2_CHILD_COUNT_BUSH_SCATTERED, // OBJMURE2_CHILDTYPE_BUSH_SCATTERED
+    OBJ_MURE2_CHILD_COUNT_ROCK_RING,      // OBJMURE2_CHILDTYPE_ROCK_RING
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_809613E8.s")
+static s16 sActorIds[] = {
+    ACTOR_EN_KUSA,
+    ACTOR_EN_KUSA,
+    ACTOR_EN_ISHI,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_809613FC.s")
+static VecPolarS sScatteredBushSpawnInfo[] = {
+    { 40, 0x0666 }, { 40, 0x2CCC }, { 40, 0x5999 }, { 40, 0x8666 }, { 20, 0xC000 }, { 80, 0x1333 },
+    { 80, 0x4000 }, { 80, 0x6CCC }, { 80, 0x9333 }, { 80, 0xACCC }, { 80, 0xC666 }, { 60, 0xE000 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_8096147C.s")
+typedef void (*ObjMure2SpawnPosFunc)(Vec3f[12], ObjMure2*);
+static ObjMure2SpawnPosFunc sSpawnPosFuncs[] = {
+    ObjMure2_GetBushCircleSpawnPos, // OBJMURE2_CHILDTYPE_BUSH_RING
+    ObjMure2_GetBushScatteredPos,   // OBJMURE2_CHILDTYPE_BUSH_SCATTERED
+    ObjMure2_GetRocksSpawnPos,      // OBJMURE2_CHILDTYPE_ROCK_RING
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_80961490.s")
+// static InitChainEntry sInitChain[] = {
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 2100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 100, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/ObjMure2_Update.s")
+void ObjMure2_GetBushCircleSpawnPos(Vec3f pos[12], ObjMure2* this) {
+    s32 i;
+
+    Math_Vec3f_Copy(&pos[0], &this->actor.world.pos);
+    for (i = 1; i < sChildCounts[OBJ_MURE2_GET_CHILD_TYPE(&this->actor)]; i++) {
+        Math_Vec3f_Copy(&pos[i], &this->actor.world.pos);
+        (pos + i)->x += 80.0f * Math_SinS((i - 1) * 0x2000);
+        (pos + i)->z += 80.0f * Math_CosS((i - 1) * 0x2000);
+    }
+}
+
+void ObjMure2_GetBushScatteredPos(Vec3f pos[12], ObjMure2* this) {
+    s32 i;
+
+    for (i = 0; i < sChildCounts[OBJ_MURE2_GET_CHILD_TYPE(&this->actor)]; i++) {
+        Math_Vec3f_Copy(pos + i, &this->actor.world.pos);
+        (pos + i)->x += sScatteredBushSpawnInfo[i].distance * Math_CosS(sScatteredBushSpawnInfo[i].angle);
+        (pos + i)->z -= sScatteredBushSpawnInfo[i].distance * Math_SinS(sScatteredBushSpawnInfo[i].angle);
+    }
+}
+
+void ObjMure2_GetRocksSpawnPos(Vec3f pos[12], ObjMure2* this) {
+    s32 i;
+
+    for (i = 0; i < sChildCounts[OBJ_MURE2_GET_CHILD_TYPE(&this->actor)]; i++) {
+        Math_Vec3f_Copy(&pos[i], &this->actor.world.pos);
+        (pos + i)->x += 80.0f * Math_SinS(i * 0x2000);
+        (pos + i)->z += 80.0f * Math_CosS(i * 0x2000);
+    }
+}
+
+void ObjMure2_GetChildParams(s16* childParams, ObjMure2* this) {
+    s32 childType;
+    s32 temp_a2;
+
+    childType = OBJ_MURE2_GET_CHILD_TYPE(&this->actor);
+    temp_a2 = (this->actor.params >> 8) & 0x1F;
```

Comment:
```suggestion
    s32 childType = OBJ_MURE2_GET_CHILD_TYPE(&this->actor);
    s32 temp_a2 = (this->actor.params >> 8) & 0x1F;
```
Can be defined where declared. Also should add a Getter for this other use of params unpacking.

Reply:
You could use `Mure2ChildType` for the `childType` too